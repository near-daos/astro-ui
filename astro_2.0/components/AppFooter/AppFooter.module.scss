@import 'styles/mixins';
@import 'styles/typography';

.root {
  align-items: flex-end;
  background-color: var(--color-brand-black);
  color: var(--color-white);
  display: flex;
  grid-area: footer;
  height: 100%;
  //padding-bottom: 60px;
  width: 100%;
}

.wrapper {
  display: grid;
  grid-template-columns: 1fr;
  margin: auto;
  width: 100%;
}

.social {
  align-items: center;
  display: flex;
  grid-area: social;
  margin-bottom: 18px;

  > a {
    margin-right: 24px;
  }
}

.icon {
  &:hover {
    color: var(--color-success);
  }
}

.report {
  grid-area: report;
}

.progress {
  display: none;
  grid-area: progress;
  margin-top: 8px;

  @include tablet-landscape {
    text-align: right;
  }
}

.repo {
  grid-area: repo;
}

.version {
  grid-area: version;
}

.devLink {
  @include font(12px, 18px, 500);
  color: var(--color-white);
  display: block;
  margin-bottom: 12px;

  &:hover {
    color: var(--color-success);
  }
}

.divider {
  font-size: 14px;
  margin: 0 4px;
}

.links {
  @include font(0.75em, 18px, 400);
  color: var(--color-neutral-60);
  column-gap: 24px;
  display: flex;
  flex-direction: column;
  grid-area: links;
  margin-top: 16px;
}

.link {
  @extend %subtitle5;
  flex-shrink: 0;
  transition: color 0.4s;
  white-space: nowrap;

  &:hover {
    color: var(--color-primary-30);
  }
}

.copyright {
  @extend %caption3;
  color: var(--color-neutral-60);
  grid-area: copyright;
  margin: 24px 0 16px;
  white-space: break-spaces;
}

.terms {
  grid-area: terms;
}

.release {
  @extend %caption3;
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  grid-area: release;
  //margin-top: 16px;

  span {
    flex-shrink: 0;
    margin-right: 6px;
    white-space: nowrap;
  }
}

.locale {
  grid-area: locale;
  margin: 16px 0 32px;
}

.bottom {
  padding: 24px;
  width: 100%;

  @include desktop {
    padding: 46px 48px 30px;
  }

  @include large-desktop {
    padding: 54px 120px 36px;
  }

  .wrapper {
    grid-template-areas:
      'invitation'
      'social'
      'report'
      'version'
      'repo'
      'links'
      'release'
      'terms'
      'copyright'
      'locale';

    max-width: 1130px;

    @include tablet-landscape {
      grid-gap: 32px;
      grid-template-areas:
        'invitation invitation invitation'
        'social social release'
        'links report copyright'
        'version . progress';
      grid-template-columns: 2fr 1fr 3fr;
    }

    @include desktop {
      grid-template-columns: 3fr 1fr 3fr;
    }
  }

  .links {
    @include tablet-landscape {
      margin-top: 0;
    }
  }

  .link {
    color: var(--color-primary-30);
  }

  .link:hover {
    color: var(--color-primary-40);
  }

  .social {
    @include tablet-landscape {
      margin-bottom: 0;
    }

    @include desktop {
      margin-bottom: 20px;
    }
  }

  .report {
    @include tablet-landscape {
      line-height: 1;
    }
  }

  .release {
    @include tablet-landscape {
      @include font(1em, 24px, 400);
      justify-self: end;
      margin-top: 0;

      @include desktop {
        margin-bottom: 20px;
      }
    }
  }

  .logo {
    @include tablet-landscape {
      width: 135px;
    }
  }

  .copyright {
    @include tablet-landscape {
      justify-self: end;
      margin-top: 0;
      text-align: right;
    }
  }

  .report,
  .version,
  .progress {
    @include tablet-landscape {
      margin-top: 0;
    }
  }
}

.side {
  bottom: 0;
  left: 0;
  padding: 16px 32px;
  width: 100%;

  @include astro-mobile {
    border-radius: 8px;
    box-shadow: 3px 2px 24px var(--color-black-opacity-1);
    height: fit-content;
  }

  .wrapper {
    grid-template-areas:
      'social'
      'report'
      'version'
      'repo'
      'links'
      'release'
      'terms'
      'copyright'
      'locale';
  }

  .progress {
    display: block;
    text-align: left;
  }
}

.mobile {
  color: var(--color-brand-black);
  padding-bottom: 0;

  .devLink,
  .copyright {
    color: var(--color-brand-black);
  }

  .icon:hover {
    color: initial;
  }

  .locale {
    margin: 16px 0;
  }
}
