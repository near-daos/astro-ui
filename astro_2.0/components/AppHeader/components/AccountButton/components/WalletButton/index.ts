export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
