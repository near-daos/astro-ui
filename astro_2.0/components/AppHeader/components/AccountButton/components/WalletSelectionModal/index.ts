export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
