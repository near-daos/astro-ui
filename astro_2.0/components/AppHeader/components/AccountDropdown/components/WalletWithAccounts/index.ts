export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
