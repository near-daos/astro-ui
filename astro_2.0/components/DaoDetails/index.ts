export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
