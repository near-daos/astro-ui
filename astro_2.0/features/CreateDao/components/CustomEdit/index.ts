import CustomEdit from './CustomEdit';

export default CustomEdit;
