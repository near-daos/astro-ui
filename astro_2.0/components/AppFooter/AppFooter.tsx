import cn from 'classnames';
import Link from 'next/link';
import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import { Icon, IconName } from 'components/Icon';
import { LocaleSelector } from 'astro_2.0/components/LocaleSelector';

import { configService } from 'services/ConfigService';

import styles from './AppFooter.module.scss';

export interface AppFooterProps {
  mobile?: boolean;
  className?: string;
  onClick?: () => void;
}

export const AppFooter: FC<AppFooterProps> = ({
  mobile,
  className,
  onClick,
}) => {
  const { t } = useTranslation();

  const { appConfig } = configService.get();

  // Note that variables on deployed envs are being controlled by dev-ops team
  const RELEASE_NOTES = appConfig?.RELEASE_NOTES || '';

  function renderSocialIcon(href: string, icon: IconName) {
    return (
      <a href={href} rel="noopener noreferrer" target="_blank">
        <Icon name={icon} width={24} className={styles.icon} />
      </a>
    );
  }

  const rootClassName = cn(styles.root, className, {
    [styles.mobile]: mobile,
  });

  return (
    <div
      tabIndex={0}
      role="button"
      onClick={onClick}
      onKeyPress={onClick}
      className={rootClassName}
    >
      <div className={styles.side}>
        <div className={styles.wrapper}>
          <div className={styles.social}>
            {renderSocialIcon(
              'https://discord.com/invite/KmMqF4KNBM',
              'socialDiscord'
            )}
            {renderSocialIcon('https://twitter.com/AstroDao', 'socialTwitter')}
            {renderSocialIcon('https://t.me/astro_near', 'socialTelegram')}
          </div>
          <div className={styles.version}>
            <a
              className={styles.devLink}
              href="https://feedback.astrodao.com"
              target="_blank"
              rel="noreferrer noopener"
            >
              {t('leaveFeedback')}
            </a>
          </div>
          <div className={styles.terms}>
            <Link passHref href="/terms-conditions">
              <a href="*" className={styles.devLink}>
                {t('termsAndConditions')}
              </a>
            </Link>
          </div>
          <div className={styles.release}>
            <a
              className={styles.devLink}
              href={RELEASE_NOTES}
              target="_blank"
              rel="noreferrer noopener"
            >
              {t('releaseNotes')}
            </a>
            <a
              className={styles.devLink}
              href="https://testnet.app.astrodao.com/"
              target="_blank"
              rel="noreferrer noopener"
            >
              {t('testnetEnv')}
            </a>
          </div>

          <div className={styles.copyright}>
            {t('components.appFooter.opensourceAsIs')}
            <br />
            {t('components.appFooter.communityDeveloped')}
          </div>

          <div className={styles.locale}>
            <LocaleSelector />
          </div>
        </div>
      </div>
    </div>
  );
};
