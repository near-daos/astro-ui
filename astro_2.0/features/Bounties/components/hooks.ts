import { useCallback, useRef, useState } from 'react';
import { useRouter } from 'next/router';
import { useAsyncFn, useLocalStorage, useLocation } from 'react-use';
import axios, { CancelTokenSource } from 'axios';

import { SputnikHttpService } from 'services/sputnik';
import { useModal } from 'components/modal';
import { ConfirmActionModal } from 'astro_2.0/components/ConfirmActionModal';

import { Bounty, BountyContext, BountyProposal } from 'types/bounties';
import { VoteAction } from 'types/proposal';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';

import { useWalletContext } from 'context/WalletContext';
import { PaginationResponse } from 'types/api';
import { VOTE_ACTION_SOURCE_PAGE } from 'constants/votingConstants';

export function useBountyControls(
  daoId: string,
  bounty?: Bounty
): {
  handleClaim: () => void;
  handleUnclaim: () => void;
  loading: boolean;
} {
  const router = useRouter();
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();
  const [loading, setLoading] = useState(false);

  const [showModal] = useModal(ConfirmActionModal);

  const onSuccessHandler = useCallback(async () => {
    router.reload();
  }, [router]);

  const handleClaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Claim',
    });

    if (res?.length && bounty) {
      setLoading(true);

      const dao = await SputnikHttpService.getDaoById(daoId);

      if (!dao) {
        return;
      }

      try {
        setVoteActionSource(pathname);

        await nearService?.claimBounty(daoId, {
          bountyId: bounty?.bountyId,
          deadline: bounty?.maxDeadline,
          bountyBond: dao.policy.bountyBond,
          gas: res[0],
          tokenId: bounty?.token,
        });

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    daoId,
    setVoteActionSource,
    pathname,
    nearService,
    onSuccessHandler,
  ]);

  const handleUnclaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Unclaim',
    });

    if (res?.length && bounty) {
      try {
        setLoading(true);
        setVoteActionSource(pathname);
        await nearService?.unclaimBounty(daoId, bounty?.bountyId);

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    setVoteActionSource,
    pathname,
    nearService,
    daoId,
    onSuccessHandler,
  ]);

  return {
    handleClaim,
    handleUnclaim,
    loading,
  };
}

export function useBountyVoting(
  daoId: string,
  proposal: BountyProposal
): {
  handleVote: (vote: VoteAction) => void;
  loading: boolean;
} {
  const router = useRouter();
  const [showModal] = useModal(ConfirmActionModal);
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();

  const [{ loading }, handleVote] = useAsyncFn(
    async (vote: VoteAction) => {
      const res = await showModal({
        title: 'Confirm Your Vote',
      });

      if (res?.length) {
        setVoteActionSource(pathname);
        await nearService?.vote(daoId, proposal.proposalId, vote, res[0]);
        await router.reload();
      }
    },
    [daoId, proposal, router, nearService, pathname, setVoteActionSource]
  );

  return {
    handleVote,
    loading,
  };
}

export function useBountySearch(): {
  handleSearch: (
    val: string
  ) => Promise<PaginationResponse<BountyContext[]> | null>;
  loading: boolean;
} {
  const router = useRouter();
  const daoId = router.query.dao as string;
  const { accountId } = useWalletContext();
  const cancelTokenRef = useRef<CancelTokenSource | null>(null);

  const [{ loading }, handleSearch] = useAsyncFn(async query => {
    if (cancelTokenRef.current) {
      cancelTokenRef.current?.cancel('Cancelled by new req');
    }

    const { CancelToken } = axios;
    const source = CancelToken.source();

    cancelTokenRef.current = source;

    return SputnikHttpService.findBountyContext({
      daoId: daoId || '',
      accountId,
      query,
      cancelToken: source.token,
    });
  }, []);

  return {
    handleSearch,
    loading,
  };
}

export function useHideBounty(): {
  handleSubmit: () => void;
  handleSelect: (id: string) => void;
  selected: string[];
  loading: boolean;
} {
  const router = useRouter();
  const showHidden = router.query?.bountyFilter === 'hidden';
  const daoId = router.query.dao as string;
  const { accountId, nearService, pkAndSignature } = useWalletContext();
  const [selected, setSelected] = useState<string[]>([]);

  const [{ loading }, handleSubmit] = useAsyncFn(async () => {
    if (!pkAndSignature) {
      return;
    }

    try {
      const { publicKey, signature } = pkAndSignature;

      if (publicKey && signature) {
        await SputnikHttpService.toggleBountyContexts({
          accountId,
          publicKey,
          signature,
          daoId,
          ids: selected,
          isArchived: !showHidden,
        });

        await router.replace(router.asPath, undefined, {
          shallow: false,
        });

        setSelected([]);
      }
    } catch (err) {
      showNotification({
        type: NOTIFICATION_TYPES.ERROR,
        lifetime: 20000,
        description: err.message,
      });
    }
  }, [
    selected,
    daoId,
    showHidden,
    accountId,
    router,
    nearService,
    pkAndSignature,
  ]);

  const handleSelect = useCallback(
    (id: string) => {
      if (selected.includes(id)) {
        setSelected(selected.filter(item => item !== id));
      } else {
        setSelected([...selected, id]);
      }
    },
    [selected]
  );

  return {
    handleSubmit,
    handleSelect,
    selected,
    loading,
  };
}
