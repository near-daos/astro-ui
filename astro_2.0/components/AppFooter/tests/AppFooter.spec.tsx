import { render } from 'jest/testUtils';

import { AppFooter } from 'astro_2.0/components/AppFooter';

describe('App Footer', () => {
  it('Should render AppFooter component', () => {
    const { container } = render(<AppFooter />);

    expect(container).toMatchSnapshot();
  });
});
