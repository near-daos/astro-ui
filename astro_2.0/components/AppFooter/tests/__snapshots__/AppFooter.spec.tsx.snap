// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Footer Should render AppFooter component 1`] = `
<div>
  <div
    class="root"
    role="button"
    tabindex="0"
  >
    <div
      class="side"
    >
      <div
        class="wrapper"
      >
        <div
          class="social"
        >
          <a
            href="https://discord.com/invite/KmMqF4KNBM"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon"
              width="24"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </a>
          <a
            href="https://twitter.com/AstroDao"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon"
              width="24"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </a>
          <a
            href="https://t.me/astro_near"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon"
              width="24"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </a>
        </div>
        <div
          class="version"
        >
          <a
            class="devLink"
            href="https://feedback.astrodao.com"
            rel="noreferrer noopener"
            target="_blank"
          >
            leaveFeedback
          </a>
        </div>
        <div
          class="terms"
        >
          <a
            class="devLink"
            href="/terms-conditions"
          >
            termsAndConditions
          </a>
        </div>
        <div
          class="release"
        >
          <a
            class="devLink"
            href="https://astrodao.canny.io/changelog"
            rel="noreferrer noopener"
            target="_blank"
          >
            releaseNotes
          </a>
          <a
            class="devLink"
            href="https://testnet.app.astrodao.com/"
            rel="noreferrer noopener"
            target="_blank"
          >
            testnetEnv
          </a>
        </div>
        <div
          class="copyright"
        >
          components.appFooter.opensourceAsIs
          <br />
          components.appFooter.communityDeveloped
        </div>
        <div
          class="locale"
        >
          <div
            class="root"
          >
            <div>
              <div
                class="selector"
              >
                <img
                  class="flag"
                  data-testid="circle-country-flag"
                  height="100"
                  src="https://hatscripts.github.io/circle-flags/flags/us.svg"
                  title="us"
                />
                 
                en
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
